guides/plugins/plugins/storefront/add-custom-controller.md

```diff
@@ -1,2 +1,216 @@
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+    /**
+    * @RouteScope(scopes={"storefront"})
+    * @Route("/example", name="frontend.example.example", methods={"GET"})
+    */
+    public function showExample(): Response
+    {
+        ...
+    }
+}
+```
+{% endcode %}
+
+### Services.xml example
+
+Next, we need to register our controller in the DI-container and make it public.
+
+{% code title="<plugin root>/src/Resources/config/services.xml" %}
+```markup
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services" 
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="Swag\BasicExample\Storefront\Controller\ExampleController" public="true"/>
+    </services>
+</container>
+```
+{% endcode %}
+
+### Routes.xml example
+
+Once we‘ve registered our new controller, we have to tell Shopware how we want it to search for new routes in our plugin.
+This is done with a `routes.xml` file at `<plugin root>/src/Resources/config/` location.
+Have a look at the official [Symfony documentation](https://symfony.com/doc/current/routing.html) about routes and how they are registered.
+
+{% code title="<plugin root>/src/Resources/config/routes.xml" %}
+```markup
+<?xml version="1.0" encoding="UTF-8" ?>
+<routes xmlns="http://symfony.com/schema/routing"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/routing
+        https://symfony.com/schema/routing/routing-1.0.xsd">
+
+    <import resource="../../Storefront/Controller/**/*Controller.php" type="annotation" />
+</routes>
+```
+{% endcode %}
+
+### Adding template
+
+Now we registered our controller and Shopware indexes the route, but the template file, that is supposed to be rendered, is still missing.
+Let's change that now.
+
+As previously mentioned, the code will try to render an `index.html.twig` file.
+Thus we have to create an `index.html.twig` in the `<plugin root>/src/Resources/views/storefront/page/example` directory, as defined in our controller.
+Below you can find an example, where we extend from the template `base.html.twig` and override the block `base_content`.
+In our [Customize templates guide](./customize-templates.md), you can learn more about customizing templates.
+
+{% code title="<plugin root>/src/Resources/views/storefront/page/example.html.twig" %}
+```text
```

Comment:
Why do some of you use \```text and some use \```twig ?
I think twig is working fine

```suggestion
```twig
```

Reply:
Afaik it only works fine because markdown defaults to text if unknown.
Twig is not a valid code markup known to default markdown, is it?

Sure, I can change that, won't break anything. Just... it will default to text anyway. What about HTML?

Reply:
@leichteckig had the same issue with using \```html, she changed it to \```twig. 
Dunno, if this matters at all.